files/upsert-records

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash -e
+
+set -o nounset
+umask 177
+
+# Parameters
+url="${1}/api/v1/servers/${2}"
+# $2 is the server
+apikey="${apikey:-}"
+zonename="${3}"
+# The rrset is passed via an env variable because the quotes are mangled when
+# specifying the JSON on the command line.
+expected="${expected:-}"
+
+# Parse zone metadata
+eval "$(echo "${expected}" | jq -r '@sh "defaultTtl=\(.defaultTTL)"')"
+
+# Filter out the records
+records="$(echo "${expected}" | jq '.records')"
+# First step of preparing the RRsets.
+# This groups the records into RRsets.
+# The TTL is at the wrong location, but step 2 fixes that
+# What is left to be done:
+# - Wrap everything into "rrsets"
+# - Add changetype if needed
+records="$(echo "${records}" | jq '[. | to_entries[] | (.value | to_entries[]) as $v | { "name": .key, "type": $v.key, "records": [ $v.value[] | if .c != null then { "content": .c, "disabled": false } + if .r != null then { "set-ptr": .r } else {} end else { "ttl": .t } end ] }]')"
```

Comment:
Doesn't `t` have to be set to be accessed here?

Reply:
Fun fact: it does